File: CoreTextDemo/CoreTextDemo/ViewController.m
//
//  ViewController.m
//  CoreTextDemo
//
//  Created by 卢良潇 on 16/4/25.
//  Copyright © 2016年 卢良潇. All rights reserved.
//

#import "ViewController.h"
#import "CTFrameParser.h"
#import "CTFrameParserConfig.h"
#import "CoreTextData.h"
#import "CTDisplayView.h"
#import "UIView+frameAdjust.h"


@interface ViewController ()
@property (weak, nonatomic) IBOutlet CTDisplayView *ctView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
    CTFrameParserConfig *config = [[CTFrameParserConfig alloc]init];
    config.textColor = [UIColor redColor];
    config.width = self.ctView.width;
    
    
    
    NSString *content = @"dfb sdouvbnwroivnwroivneroivneriovneriovneornvernveaoivnesoivnesoivnesoinveionveoinveorinveroivneroinverionvrieonvoirnvoiernvoiernviore";
    NSDictionary *attr = [CTFrameParser attributesWithConfig:config];
    NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithString:content attributes:attr];
    [attributedString addAttribute:NSForegroundColorAttributeName value:[UIColor redColor] range:NSMakeRange(0, 7)];
    CoreTextData *data = [CTFrameParser parseContent:@"我来试试11111111111111111111111111" config:config];
    
    
    
    self.ctView.data = data;
    self.ctView.height = data.height;
    self.ctView.backgroundColor = [UIColor yellowColor];
    
    
    
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: HeartAnimation/HeartAnimation/ViewController.swift
//
//  ViewController.swift
//  HeartAnimation
//
//  Created by 卢良潇 on 16/7/16.
//  Copyright © 2016年 卢良潇. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.backgroundColor = UIColor.white()
        let imageView = UIImageView(frame: view.bounds)
        imageView.image = UIImage(named: "bgview")
        view.addSubview(imageView)
        
        Timer.scheduledTimer(timeInterval: 0.25, target: self, selector: #selector(ViewController.showHeartView), userInfo: nil, repeats: true)
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    
     func showHeartView(){
    
        let heartView = HeartView(frame: CGRect(x: 0, y: 0, width: 40, height: 40))
        view.addSubview(heartView)
        let fountainSource = CGPoint(x: view.frame.size.width - 80, y: view.bounds.size.height - 30 / 2.0 - 10)
        heartView.center = fountainSource
        heartView.animateInView(view: view)
    }
    

}

